### PR TITLE
[CCR] Make auto follow patterns work with security

### PR DESCRIPTION
**Pull request by martijnvg**
Relates to #33007

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/FollowIndexAction.java [hunk 1]
```diff
@@ -404,7 +404,9 @@ void start(
             final AtomicReferenceArray<Object> responses = new AtomicReferenceArray<>(followIndexMetadata.getNumberOfShards());
             Map<String, String> filteredHeaders = threadPool.getThreadContext().getHeaders().entrySet().stream()
                 .filter(e -> ShardFollowTask.HEADER_FILTERS.contains(e.getKey()))
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));for (int i = 0; i < numShards; i++) {
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
Comment (martijnvg) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/FollowIndexAction.java]: just added a new lines here for readability.

**Comment by jasontedor** [comment]
@martijnvg Would you merge master into this and resolve the conflicts?

**Comment by martijnvg** [comment]
@jasontedor I merged it in tomorrow, but I forgot to push.

**Comment by martijnvg** [comment]
Jenkins, run sample packaging tests

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrLicenseChecker.java [hunk 2]
```diff
@@ -105,12 +116,14 @@ public boolean isCcrAllowed() {
      */
     public <T> void checkRemoteClusterLicenseAndFetchClusterState(
             final Client client,
+            final Map<String, String> headers,
```
[the remark this answers is not in the document]
Comment (jasontedor) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrLicenseChecker.java]: As we discussed offline, this is okay for now, we can always refactor this later. The implementation is fine, we just want to take a step back and look at how ML uses the remote license checker too (multiple clusters) which has kind of forced how we implement this.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**